app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoFragment.kt

```diff
@@ -53,31 +56,38 @@ class PocketVideoFragment : Fragment() {
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        fun displayFeed(layout: View, videos: PocketVideosDeferred) {
-            layout.videoFeed.gridView.adapter = PocketVideoAdapter(context!!, videos, fragmentManager!!)
-        }
-
-        val layout = inflater.inflate(R.layout.fragment_pocket_video, container, false)
+        return inflater.inflate(R.layout.fragment_pocket_video, container, false)
+    }
 
-        if (deferredVideos.isCompleted) {
-            displayFeed(layout, deferredVideos)
-        } else {
-            // TODO: #769?: display loading screen.
-
-            launch(UI) {
-                val videos = deferredVideos.await()
-                if (videos != null) {
-                    displayFeed(layout, deferredVideos)
-                } else {
-                    // TODO: #769: display error screen.
-                }
+    override fun onStart() {
+        super.onStart()
+        val adapter = PocketVideoAdapter(context!!, fragmentManager!!).also {
+            videoFeed.gridView.adapter = it
+        }
+        fun setupLoadingIfApplicable() {
+            if (deferredVideos.isNotCompleted()) {
+                // We show 20 placeholders because our feed will be updated to show 20 items
+                val placeholders = List(20) { PocketPlaceholder }
+                adapter.setVideos(placeholders)
             }
         }
+        fun displayFeedOrError(videos: List<PocketVideo>?) {
+            if (videos != null) {
+                adapter.setVideos(videos)
+            } else {
+                // TODO: #769: display error screen.
+            }
+        }
+
+        launch(UI + uiLifecycleCancelJob) {
```

Comment:
nit: `launch(UI + uiLifecycleCancelJob, UNDISPATCHED) {` It's slightly more efficient because we don't have to dispatch to the UI thread, though it will mean the flow is more likely to be different between loading and finished (i.e. **Currently:** SUSPEND loading SUSPEND feed / SUSPEND feed **vs.** loading SUSPEND feed / feed).